File: eCar/eCar.Services/Database/Rent.cs
using System;
using System.Collections.Generic;

namespace eCar.Services.Database;

public partial class Rent
{
    public int Id { get; set; }

    public DateTime? RentingDate { get; set; }

    public DateTime? EndingDate { get; set; }

    public int? NumberOfDays { get; set; }

    public decimal? FullPrice { get; set; }
    public bool? Paid { get; set; }

    public string? Status { get; set; }

    public int VehicleId { get; set; }

    public int ClientId { get; set; }

    public virtual Client Client { get; set; } = null!;

    public virtual Vehicle Vehicle { get; set; } = null!;
}

File: eCar/UI/ecar_mobile/lib/models/Rent/rent.dart
import 'package:ecar_mobile/models/Client/client.dart';
import 'package:ecar_mobile/models/Vehicle/vehicle.dart';
import 'package:json_annotation/json_annotation.dart';

part 'rent.g.dart';

@JsonSerializable()
class Rent {
  int? id;
  DateTime? rentingDate;
  DateTime? endingDate;
  int? numberOfDays;
  double? fullPrice;
  bool? paid;
  String? status;
  int? vehicleId;
  int? clientId;
  Client? client;
  Vehicle? vehicle;

  Rent(
      {this.id,
      this.rentingDate,
      this.endingDate,
      this.numberOfDays,
      this.fullPrice,
      this.paid,
      this.status,
      this.vehicleId,
      this.clientId,
      this.client,
      this.vehicle});

  factory Rent.fromJson(Map<String, dynamic> json) => _$RentFromJson(json);

  Map<String, dynamic> toJson() => _$RentToJson(this);
}
